docs/contract_types.rst

```diff
@@ -324,6 +324,49 @@ Note: you are not allowed to mix different kinds of separators on the same line.
         mypackage.low
 
 
+Protected modules
+-----------------
+
+*Type name:* ``protected``
+
+Protected contracts check that one set of modules is only imported by another set of modules.
+
+By default, descendants of each module will be checked - so if ``mypackage.one`` is protected, with ``mypackage.two`` 
+marked as allow to import it, then ``mypackage.two.blue`` can import ``mypackage.one.green`` when ``mypackage.three.red``
+cannot. This descendant behaviour can be changed by setting ``as_packages`` to ``False``: in that case, only explicitly
+listed modules will be checked, not their descendants.
+
+**Examples:**
+
+.. code-block:: ini
+
+    [importlinter]
+    root_package = mypackage
+
+    [importlinter:contract:my-protected-contract]
+    name = My protected contract
+    type = protected
+    protected_modules =
+        mypackage.one
+    allowed_importers =
+        mypackage.two
+        mypackage.three.blue
+    ignore_imports = 
```

Comment:
Could we remove the `ignore_imports` (so the example is more minimal) and then add another example contract with all of the optional flags? Would also be good to show wildcards in that second example.